aiohomekit/controller/ip/connection.py

```diff
@@ -280,8 +282,15 @@ async def ensure_connection(self):
         Otherwise, start a reconnection and wait for it.
         """
         if self._start_reconnecting():
+            # If we are running under a timeout, we still need to shield the
+            # connector task so it continues to run if the timeout is hit.
             await asyncio.shield(self._connector)
 
+        if self._last_connector_error:
+            raise self._last_connector_error
```

Comment:
We now raise the last connection error if there was a problem connecting since all we ever told the user was it timed out otherwise